### PR TITLE
Support for SAP Hana Database

### PR DESCRIPTION
**Pull request by MichaelSp**
The driver itself can be found here:

```xml
<dependency>
    <groupId>com.sap.db.jdbc</groupId>
    <artifactId>ngdbc</artifactId>
</dependency>
```

### CODE REVIEW

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/jdbc/DatabaseDriver.java [hunk 1]
```diff
@@ -16,14 +16,14 @@
 
 package org.springframework.boot.jdbc;
 
+import org.springframework.util.Assert;
```
Comment (snicoll) [inline spring-boot-project/spring-boot/src/main/java/org/springframework/boot/jdbc/DatabaseDriver.java]: This layout change seems unrelated. Could you please revert that part?

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/jdbc/DatabaseDriver.java [hunk 2]
```diff
@@ -99,6 +99,18 @@ public String getId() {
 	POSTGRESQL("PostgreSQL", "org.postgresql.Driver", "org.postgresql.xa.PGXADataSource",
 			"SELECT 1"),
 
+	/**
+	 * SAP - SAP Hana Database - HDB.
+	 */
+    HDB("HDB", "com.sap.db.jdbc.Driver", "com.sap.db.jdbcext.XADataSourceSAP",
```
[the remark this answers is not in the document]
Comment (MichaelSp) [reply spring-boot-project/spring-boot/src/main/java/org/springframework/boot/jdbc/DatabaseDriver.java]: The driver itself has this:
```
    public static String getDatabaseName() {
        return "HDB";
    }

    public static String getDriverName() {
        return "HDB";
    }
```

I don't know the implications, but to me it looks like the name "HDB" aligns better. Open for discussion....

**Comment by MichaelSp** [comment]
arghs... https://stackoverflow.com/questions/42431906/ngdbc-jar-equivalent-maven-repository 
any idea?

**Comment by snicoll** [comment]
You shouldn't add a reference to the driver as we're not offering dependency management for it. It would be nice if you add a single proposal rather than opening a PR and then amending things after the fact. 

Please remove the driver in the pom, it is not needed.

**Comment by MichaelSp** [comment]
Ok, sorry. I thought the change is straight forward...

**Comment by snicoll** [comment]
@MichaelSp we've just discussed this and we think `SAP` is a better name for the enum. Can you please restore your proposal to use that name?
